src/main/java/www/purple/mixxy/controllers/LoginLogoutController.java

```diff
@@ -134,11 +134,20 @@ public Result validateGoogleAuth(String provider, String code, Context context)
 				ObjectifyProvider objectifyProvider = new ObjectifyProvider();
 		        Objectify ofy = objectifyProvider.get();
 		        
-		        // Create a new user and save it
-		        User user = new User(userdata.getEmail(), userdata.getGiven_name(), userdata.getFamily_name(), userdata.getEmail(),
-		        		userdata.getPicture(), userdata.getLocale(), provider, userdata.getId());
-		        ofy.save().entity(user).now();
+		        // Retrieve the user with e-mail address
+		        User user = ofy.load().type(User.class).filter("username", userdata.getEmail()).first().now();
 		        
+		        //	Check if the User already exists in the Datastore
+		        //	if not, create that user, otherwise use the User instance
+		        if(user == null) {
```

Comment:
There's probably some way we can encapsulate this behavior so we don't have to keep writing it.  But we'll leave that for later.